opm/simulators/utils/readDeck.cpp

```diff
@@ -448,14 +451,68 @@ void Opm::ensureOutputDirExists(const std::string& cmdline_output_dir)
 void Opm::prepareResultOutputDirectory(const std::string&           baseName,
                                        const std::filesystem::path& outputDir)  
 {
-    //Loop over all files in the output directory and subdirectories and delete them if their name is baseName + a correct extension
-    std::regex r(baseName + R"(\.(F?(DBG|E?GRID|INIT|PRT|RFT|SMSPEC|UNSMRY|UNRST)|([ABCFGHSTUXYZ]\d{4})|(INFOSTEP|INFOITER|OPMRST)))");
-    for (auto& file : std::filesystem::directory_iterator(outputDir)) {
-        std::string fileName = file.path().filename();
-        if (std::regex_match(fileName, r)) {
-            std::filesystem::remove(file);
-        }
-    }
+    namespace fs = std::filesystem;
+
+    // Delete every regeular file in the output directory whose name is
```

Comment:
regular

Reply:
> regular

Thanks.  I've pushed an update to correct the spelling.  I've also switched to using `std::for_each()` for the directory traversal as well.